### ISSUE TITLE
generate_change_report() fails with PermissionError

### ISSUE THREAD

**Issue by dbantje** [polca/premise#163]
With the new 2.1.0.dev4 version, `generate_change_report()` called from within `write_db_to_rbightway` fails with the following error message:

```
---------------------------------------------------------------------------
PermissionError                           Traceback (most recent call last)
Cell In[10], [line 2](vscode-notebook-cell:?execution_count=10&line=2)
      [1](vscode-notebook-cell:?execution_count=10&line=1) for scen in SCENARIOS:
----> [2](vscode-notebook-cell:?execution_count=10&line=2)     create_inventories(scen, YEARS, EI_VERSION)

Cell In[8], [line 43](vscode-notebook-cell:?execution_count=8&line=43)
     [41](vscode-notebook-cell:?execution_count=8&line=41)     ndb.update("two_wheelers")
     [42](vscode-notebook-cell:?execution_count=8&line=42)     ndb.update("cars")
---> [43](vscode-notebook-cell:?execution_count=8&line=43) ndb.write_db_to_brightway()
     [44](vscode-notebook-cell:?execution_count=8&line=44) print("Creation of scenario {} took: {:.1f} min"
     [45](vscode-notebook-cell:?execution_count=8&line=45)         .format(scenario, (time.time() - start_time)/60.))
     [46](vscode-notebook-cell:?execution_count=8&line=46) del ndb

File [c:\Users\davidba\.conda\envs\costs_paper\lib\site-packages\premise\new_database.py:1027](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1027), in NewDatabase.write_db_to_brightway(self, name)
   [1025](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1025) self.generate_scenario_report()
   [1026](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1026) # generate change report from logs
-> [1027](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1027) self.generate_change_report()

File [c:\Users\davidba\.conda\envs\costs_paper\lib\site-packages\premise\new_database.py:1239](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1239), in NewDatabase.generate_change_report(self)
   [1234](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1234) """
   [1235](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1235) Generate a report of the changes between the original database and the scenarios.
   [1236](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1236) """
   [1238](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1238) print("Generate change report.")
-> [1239](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1239) generate_change_report(
   [1240](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1240)     self.source, self.version, self.source_type, self.system_model
   [1241](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1241) )
   [1242](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1242) # saved under working directory
   [1243](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/new_database.py:1243) print(f"Report saved under {os.getcwd()}.")

File [c:\Users\davidba\.conda\envs\costs_paper\lib\site-packages\premise\report.py:619](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/report.py:619), in generate_change_report(source, version, source_type, system_model)
    [615](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/report.py:615) fp = Path(
    [616](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/report.py:616)     DIR_LOG_REPORT / f"change_report {datetime.now().strftime('%Y-%m-%d')}.xlsx"
    [617](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/report.py:617) )
    [618](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/report.py:618) workbook.save(fp)
--> [619](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/report.py:619) empty_log_files()

File [c:\Users\davidba\.conda\envs\costs_paper\lib\site-packages\premise\logger.py:61](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/logger.py:61), in empty_log_files()
     [58](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/logger.py:58) for file in DIR_LOG_REPORT.iterdir():
     [59](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/logger.py:59)     # if suffix is ".log"
     [60](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/logger.py:60)     if file.suffix == ".log":
---> [61](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/site-packages/premise/logger.py:61)         file.unlink()

File [c:\Users\davidba\.conda\envs\costs_paper\lib\pathlib.py:1206](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1206), in Path.unlink(self, missing_ok)
   [1201](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1201) """
   [1202](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1202) Remove this file or link.
   [1203](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1203) If the path is a directory, use rmdir() instead.
   [1204](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1204) """
   [1205](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1205) try:
-> [1206](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1206)     self._accessor.unlink(self)
   [1207](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1207) except FileNotFoundError:
   [1208](file:///C:/Users/davidba/.conda/envs/costs_paper/lib/pathlib.py:1208)     if not missing_ok:

PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\Users\\davidba\\costs_paper\\for_publication\\scripts\\export\\logs\\premise_biomass.log'
```

**Comment by romainsacchi** [comment]
OK.
BTW, now `.update()` includes everything, including "cars", "two_wheelers", "trucks", etc.
Can you write the exact script so that I try on my end to reproduce thsi error?

**Comment by dbantje** [comment]
```
RUNS_PATH = "../data"

def get_shorter_version_number(ei_version):
    splits = ei_version.split(".")
    if len(splits) == 2:
        return ei_version
    if len(splits) > 2:
        return ".".join(splits[0:2])

def create_inventories(scenario, years, ei_version, update_pers_transport=False):
    """
    Create REMIND-ecoinvent inventories for given years and
    scenario. Inventories for scenarios that are to be
    omitted are deleted from the database to speed up
    the database.
    """
    ei_label = "ecoinvent {} cutoff".format(ei_version)

    print("Creating inventories for {}.".format(scenario))

    dblist = [db for db in bw.databases]

    for year in years:

        scen = {"model": "remind", "pathway": scenario, "year": year, "filepath": RUNS_PATH}
        dbname = eidb_label(scen, ei_version)
        if dbname in dblist:
            print("Existing database found for {}, year {}.".format(scenario, year))
            continue
        
        start_time = time.time()
        ndb = NewDatabase(scenarios=[scen],
                            source_db=ei_label,
                            source_version=ei_version,
                            keep_uncertainty_data=True
                            # key="tUePmX_S5B8ieZkkM7WUU2CnO8SmShwmAeWK9x2rTFo=")
        )
        ndb.update()
        if update_pers_transport:
            ndb.update("buses")
            ndb.update("two_wheelers")
            ndb.update("cars")
        ndb.write_db_to_brightway()
        print("Creation of scenario {} took: {:.1f} min"
                .format(scenario, (time.time() - start_time)/60.))
        del ndb
        gc.collect()
```

**Comment by dbantje** [comment]
Fixed now with version 2.1.0.dev5. Thank you!
